### PR TITLE
perf(lttng): improve memory consumption

### PR DESCRIPTION
**Pull request by hsgwa**
Signed-off-by: hsgwa <hasegawa.isp@gmail.com>

This PR improves processing time and memory usage for Lttng class.

For now, I removed ros2_tracing classes.
memory usage was improved, but processing time was worse than with cache.

[Without event filter]

Before (without cache)
15.8s, 333MB

Before (with cache)
 6.3s, 335MB

After
18.5s, 102MB

[With event filter]
Before (without cache)
13.2s, 274MB

Before(with cache)
3.84s, 245MB

After
13.9 s, 23MB

### CODE REVIEW

File: src/test/infra/lttng/test_architecture_reader_lttng.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2021 Research Institute of Systems Planning, Inc.
+# Copyright 2021 esearch Institute of Systems Planning, Inc.
```
Comment (takam5f2) [inline src/test/infra/lttng/test_architecture_reader_lttng.py:1]:
```suggestion
# Copyright 2021 Research Institute of Systems Planning, Inc.
```

**Comment by takam5f2** [comment]
@hsgwa  
I found a failure on generating an architecture file with `Architecture('lttng', observed_data_path)` method.
Have you ever seen a similar failure?

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/tmp/ipykernel_147068/431793165.py in <module>
----> 1 arch.export('x2_architecture_galactic.yaml', force=True)

~/caret/build/caret_analyze/caret_analyze/architecture/architecture.py in export(self, file_path, force)
    147         exporter = ArchitectureExporter(
    148             self.nodes, self.executors, self.paths, force)
--> 149         exporter.execute(file_path)
    150 
    151     def search_paths(

~/caret/build/caret_analyze/caret_analyze/architecture/architecture_exporter.py in execute(self, file_path)
     44         mode = 'w' if self._force else 'x'
     45         with open(file_path, mode=mode) as f:
---> 46             f.write(str(self))
     47 
     48     def __str__(self) -> str:

~/caret/build/caret_analyze/caret_analyze/architecture/architecture_exporter.py in __str__(self)
     49         import yaml
     50         obj = self.to_dict()
---> 51         return yaml.dump(obj, indent=2, default_flow_style=False, sort_keys=False)
     52 
     53     def to_dict(self):

/usr/local/lib/python3.10/dist-packages/yaml/__init__.py in dump(data, stream, Dumper, **kwds)
    251     If stream is None, return the produced string instead.
    252     """
--> 253     return dump_all([data], stream, Dumper=Dumper, **kwds)
    254 
    255 def safe_dump_all(documents, stream=None, **kwds):

/usr/local/lib/python3.10/dist-packages/yaml/__init__.py in dump_all(documents, stream, Dumper, default_style, default_flow_style, canonical, indent, width, allow_unicode, line_break, encoding, explicit_start, explicit_end, version, tags, sort_keys)
    239         dumper.open()
    240         for data in documents:
--> 241             dumper.represent(data)
    242         dumper.close()
    243     finally:

/usr/local/lib/python3.10/dist-packages/yaml/representer.py in represent(self, data)
     25 
     26     def represent(self, data):
---> 27         node = self.represent_data(data)
     28         self.serialize(node)
     29         self.represented_objects = {}

/usr/local/lib/python3.10/dist-packages/yaml/representer.py in represent_data(self, data)
     46         data_types = type(data).__mro__
     47         if data_types[0] in self.yaml_representers:
---> 48             node = self.yaml_representers[data_types[0]](self, data)
     49         else:
     50             for data_type in data_types:

/usr/local/lib/python3.10/dist-packages/yaml/representer.py in represent_dict(self, data)
    205 
    206     def represent_dict(self, data):
--> 207         return self.represent_mapping('tag:yaml.org,2002:map', data)
    208 
    209     def represent_set(self, data):

/usr/local/lib/python3.10/dist-packages/yaml/representer.py in represent_mapping(self, tag, mapping, flow_style)
    116         for item_key, item_value in mapping:
    117             node_key = self.represent_data(item_key)
--> 118             node_value = self.represent_data(item_value)
    119             if not (isinstance(node_key, ScalarNode) and not node_key.style):
    120                 best_style = False

/usr/local/lib/python3.10/dist-packages/yaml/representer.py in represent_data(self, data)
     46         data_types = type(data).__mro__
     47         if data_types[0] in self.yaml_representers:
---> 48             node = self.yaml_representers[data_types[0]](self, data)
     49         else:
     50             for data_type in data_types:

/usr/local/lib/python3.10/dist-packages/yaml/representer.py in represent_list(self, data)
    197         #            break
    198         #if not pairs:
--> 199             return self.represent_sequence('tag:yaml.org,2002:seq', data)
    200         #value = []
    201         #for item_key, item_value in data:

/usr/local/lib/python3.10/dist-packages/yaml/representer.py in represent_sequence(self, tag, sequence, flow_style)
     90         best_style = True
     91         for item in sequence:
---> 92             node_item = self.represent_data(item)
     93             if not (isinstance(node_item, ScalarNode) and not node_item.style):
     94                 best_style = False

/usr/local/lib/python3.10/dist-packages/yaml/representer.py in represent_data(self, data)
     46         data_types = type(data).__mro__
     47         if data_types[0] in self.yaml_representers:
---> 48             node = self.yaml_representers[data_types[0]](self, data)
     49         else:
     50             for data_type in data_types:

/usr/local/lib/python3.10/dist-packages/yaml/representer.py in represent_dict(self, data)
    205 
    206     def represent_dict(self, data):
--> 207         return self.represent_mapping('tag:yaml.org,2002:map', data)
    208 
    209     def represent_set(self, data):

/usr/local/lib/python3.10/dist-packages/yaml/representer.py in represent_mapping(self, tag, mapping, flow_style)
    116         for item_key, item_value in mapping:
    117             node_key = self.represent_data(item_key)
--> 118             node_value = self.represent_data(item_value)
    119             if not (isinstance(node_key, ScalarNode) and not node_key.style):
    120                 best_style = False

/usr/local/lib/python3.10/dist-packages/yaml/representer.py in represent_data(self, data)
     46         data_types = type(data).__mro__
     47         if data_types[0] in self.yaml_representers:
---> 48             node = self.yaml_representers[data_types[0]](self, data)
     49         else:
     50             for data_type in data_types:

/usr/local/lib/python3.10/dist-packages/yaml/representer.py in represent_list(self, data)
    197         #            break
    198         #if not pairs:
--> 199             return self.represent_sequence('tag:yaml.org,2002:seq', data)
    200         #value = []
    201         #for item_key, item_value in data:

/usr/local/lib/python3.10/dist-packages/yaml/representer.py in represent_sequence(self, tag, sequence, flow_style)
     90         best_style = True
     91         for item in sequence:
---> 92             node_item = self.represent_data(item)
     93             if not (isinstance(node_item, ScalarNode) and not node_item.style):
     94                 best_style = False

/usr/local/lib/python3.10/dist-packages/yaml/representer.py in represent_data(self, data)
     46         data_types = type(data).__mro__
     47         if data_types[0] in self.yaml_representers:
---> 48             node = self.yaml_representers[data_types[0]](self, data)
     49         else:
     50             for data_type in data_types:

/usr/local/lib/python3.10/dist-packages/yaml/representer.py in represent_dict(self, data)
    205 
    206     def represent_dict(self, data):
--> 207         return self.represent_mapping('tag:yaml.org,2002:map', data)
    208 
    209     def represent_set(self, data):

/usr/local/lib/python3.10/dist-packages/yaml/representer.py in represent_mapping(self, tag, mapping, flow_style)
    116         for item_key, item_value in mapping:
    117             node_key = self.represent_data(item_key)
--> 118             node_value = self.represent_data(item_value)
    119             if not (isinstance(node_key, ScalarNode) and not node_key.style):
    120                 best_style = False

/usr/local/lib/python3.10/dist-packages/yaml/representer.py in represent_data(self, data)
     50             for data_type in data_types:
     51                 if data_type in self.yaml_multi_representers:
---> 52                     node = self.yaml_multi_representers[data_type](self, data)
     53                     break
     54             else:

/usr/local/lib/python3.10/dist-packages/yaml/representer.py in represent_object(self, data)
    340         if not args and not listitems and not dictitems \
    341                 and isinstance(state, dict) and newobj:
--> 342             return self.represent_mapping(
    343                     'tag:yaml.org,2002:python/object:'+function_name, state)
    344         if not listitems and not dictitems  \

/usr/local/lib/python3.10/dist-packages/yaml/representer.py in represent_mapping(self, tag, mapping, flow_style)
    116         for item_key, item_value in mapping:
    117             node_key = self.represent_data(item_key)
--> 118             node_value = self.represent_data(item_value)
    119             if not (isinstance(node_key, ScalarNode) and not node_key.style):
    120                 best_style = False

/usr/local/lib/python3.10/dist-packages/yaml/representer.py in represent_data(self, data)
     50             for data_type in data_types:
     51                 if data_type in self.yaml_multi_representers:
---> 52                     node = self.yaml_multi_representers[data_type](self, data)
     53                     break
     54             else:

/usr/local/lib/python3.10/dist-packages/yaml/representer.py in represent_object(self, data)
    315             reduce = copyreg.dispatch_table[cls](data)
    316         elif hasattr(data, '__reduce_ex__'):
--> 317             reduce = data.__reduce_ex__(2)
    318         elif hasattr(data, '__reduce__'):
    319             reduce = data.__reduce__()

TypeError: cannot pickle 'SwigPyObject' object
```

**Comment by hsgwa** [comment]
I reproduced the same error in my environment.
[6cc3883](https://github.com/tier4/CARET_analyze/pull/100/commits/6cc3883ec594374cbae6b411e7f6151320c10f24) fixes this error.
Could you try again after applying [6cc3883](https://github.com/tier4/CARET_analyze/pull/100/commits/6cc3883ec594374cbae6b411e7f6151320c10f24)?
> @hsgwa I found a failure on generating an architecture file with `Architecture('lttng', observed_data_path)` method. Have you ever seen a similar failure? Otherwise, do I have to rebase this branch by main branch?

**Comment by hsgwa** [comment]
@takam5f2 

I reproduced the same error in my environment.
5075445 fixes this error.
Could you try again after applying 5075445?
> I found a failure on generating an architecture file with `Architecture('lttng', observed_data_path)` method. Have you ever seen a similar failure? Otherwise, do I have to rebase this branch by main branch?

**Review by takam5f2 — DISMISSED** [review]
@hsgwa  
Thank you for fixing bug. I can generate an architecture file, message flow and callback latency graph. 
[caret_evaluation.zip](https://github.com/tier4/CARET_analyze/files/9367074/caret_evaluation.zip)

**Comment by takam5f2** [comment]
I have checked memory consumption with about 5 minutes trace file. With this PR, memory consumption will be decrease to less than 30% of original one.

- before applying the PR
  - peak memory: 6447.66 MiB, increment: 6300.49 MiB
- after applying the PR
  - peak memory: 2187.73 MiB, increment: 2078.02 MiB

**Comment by hsgwa** [comment]
@takam5f2 
Thank you for evaluating!

I've implemented cache.
It's supposed that almost time consuming is as mush as main branch.

```
before implementing cache (57903aa)
308Mib 2m17s

after implementing cache
[without cache] ("caret_converted" will be created)
386Mib 2m10ss

[with cache] ("caret_converted" will be used)
386Mib 40s
```

I confirmed the effects of the cache.
Could you evaluate performance again?

**Comment by takam5f2** [comment]
@hsgwa  
Thank you for improving speed of reading data. 

I confirmed that cache file is reasonable to use and I confirmed that message_flow is generated.

first
```
peak memory: 7239.36 MiB, increment: 7093.11 MiB
CPU times: user 6min 58s, sys: 3.46 s, total: 7min 1s
Wall time: 7min 1s
```

seconds
```
peak memory: 7118.35 MiB, increment: 6972.65 MiB
CPU times: user 1min 28s, sys: 1.94 s, total: 1min 30s
Wall time: 1min 31s
```

**Review by takam5f2 — APPROVED** [review]

**Pull request merged**